src/core/lib/surface/call.c

```diff
@@ -427,6 +429,8 @@ static void destroy_call(grpc_exec_ctx *exec_ctx, void *call,
 
   get_final_status(call, set_status_value_directly,
                    &c->final_info.final_status);
+  c->final_info.stats.latency =
+      gpr_time_sub(gpr_now(GPR_CLOCK_MONOTONIC), c->start_time);
 
```

Comment:
I generally prefer to avoid assertions whenever possible, because it's always better for an application to return an error than to crash.  And in this particular case, the assertion wouldn't even add any benefit, because nothing would break if it's not true -- we'd simply get a bogus latency value returned.